toc/rfc/rfc-0007-repository-ownership.md

```diff
@@ -55,24 +55,94 @@ The TOC MUST also approve of any proposal to create a new repository within the
+2. Working Group A, Working Group B, and TOC approve the PR.
+3. The TOC merges the PR.
+
+#### Case 4: A Working Group wants to archive a repository that it currently owns.
+
+Approval requirements
+* The Working Group MUST approve the removal of the repository from the set of repositories that it owns.
+* The TOC MUST approve the removal of the repository from the Working Group and the archival.
+
+Steps
+1. Someone opens a PR that: marks the repo as archived in the [cloudfoundry.yml](../../org/cloudfoundry.yml) and that removes the repo from the Working Group's yaml.
+2. The Working Group and TOC both approve the PR.
+3. The TOC merges the PR.
+
+#### Case 5: A Working Group wants to delete a repository that it currently owns.
+
+Approval requirements
+* The Working Group MUST approve the removal of the repository from the set of repositories that it owns.
+* The TOC MUST approve the removal of the repository from the Working Group and the deletion.
+
+Steps
+1. Someone opens a PR that removes the [cloudfoundry.yml](../../org/cloudfoundry.yml) and removes the repo from the Working Group's yaml.
+2. The Working Group and TOC both approve the PR.
+3. The TOC merges the PR.
+4. A github admin deletes the repo.
+
+#### Case 6: A Working Group wants to cease maintaining a repository that it currently owns.
+
+Approval requirements
+* The Working Group MUST approve the removal of the repository from the set of repositories that it owns.
+* The TOC MUST approve the removal of the repository from the Working Group.
+* In addition, the TOC MUST decide whether to archive the repository or to arrange for another Working Group to own it instead (possibly a newly formed one, if appropriate and if supported by community contributors).
+* Note: In this case, it is unlikely that either the TOC or the CFF staff should own this repository, as it was already owned by some Working Group in the CF community.
+
+
+Steps
+1. Someone opens a PR that removes the repo from the Working Group's yaml.
+2. The TOC reviews and decides what action to take with the repo. Other PRs might be required for this step.
+3. The Working Group and TOC both approve the PR.
+5. The TOC merges the PR.
+
+#### Case 7: A Working Group wants to rename a repository that it currently owns.
+
+Approval requirements
+* The Working Group A and the TOC MUST both approve of the new name for the repository.
+
+Steps
+1. Someone two PRs:
+   * PR 1: Add the new repo to the [cloudfoundry.yml](../../org/cloudfoundry.yml).
+   * PR 2: Rename the repo in the Working Group's yaml. Remove the old repo from the [cloudfoundry.yml](../../org/cloudfoundry.yml).
+The second PR adds the repo to the desired Working Group's yaml. These PRs should link to each other.
+3. The Working Group and TOC both approve both PRs.
+4. The TOC merges the first PR.
+5. Wait for automation to run to create the repo.
+6. A github admin copies the old repo contents to the new repo.
```

Comment:
This renaming procedure seem risky. The org automation takes a long time so that there is easily a day between copying the repo content and publishing the new repo / deleting the old repo. Commits to the old repo after copying may get lost. Commits to the old repo are possible until the org automation has applied the second PR.

Maybe the old repo should be archived after copying immediately by the admin to reduce this risk. But it gets unarchived again by the next org automation run (before PR 2 is merged).

A possible alternative process:
- just one PR that renames the repo in cloudfoundy.yml and the WG yaml (note: this is something that the org automation can't handle correctly)
- WG and TOC approve the PR
- TOC stops org automation
- TOC merges PR
- TOC [renames the repo using the github UI](https://docs.github.com/en/repositories/creating-and-managing-repositories/renaming-a-repository) 
- TOC re-enables the org automation

Disclaimer: Haven't tested this procedure. If this manual procedure is not applied correctly, the org automation should fail (team assignment to non-existing repo).

Reply:
This approach sounds more straight forward. Should a running org automation job be canceled or just preventing the next run is OK?

Reply:
I would ensure that the org automation is not not running, neither with the an old commit nor with the merged rename PR. Otherwise the org automation may create unwanted empty repos that interfere with the rename. I guess that a running old workflow run would fail most of the time due to a missing repo - but with bad timing it may recreate the old repo.

A safe approach would be to disable the workflow and wait until a running org automation stops or, if its urgent, cancel a running automation.

Reply:
I like this approach. I will update the text.